x/ecocredit/server/basket/put.go

```diff
@@ -2,11 +2,181 @@ package basket
 
 import (
 	"context"
+	"time"
 
+	"github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	basketv1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/basket/v1"
+	regenmath "github.com/regen-network/regen-ledger/types/math"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
 	baskettypes "github.com/regen-network/regen-ledger/x/ecocredit/basket"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-func (k Keeper) Put(ctx context.Context, put *baskettypes.MsgPut) (*baskettypes.MsgPutResponse, error) {
-	//TODO implement me
-	panic("implement me")
+func (k Keeper) Put(ctx context.Context, req *baskettypes.MsgPut) (*baskettypes.MsgPutResponse, error) {
+	ownerAddr, _ := sdk.AccAddressFromBech32(req.Owner)
```

Comment:
must check error

Reply:
wouldn't that get caught in validate basic?

Reply:
i guess but still...